Automattic/jetpack
# Add Autoloader debugger helper.

This PR adds a new module to the Debug Helper plugin that provides information about current Autoloader data. Most of it is something we can see, but it's good to have it on one page. Something else that this module does is get all classmaps and filemaps and combines them to see what files are defined for each class by every active plugin. It even checks if there are unreadable files among them and emits a warning if there are some.

Suggestions for useful data are welcome!

#### Changes proposed in this Pull Request:
* Adds the Autoloader debugger module to the Debug Helper plugin.

#### Jetpack product discussion


* N/A

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:





* Go to the Jetpack Debug tools page: /wp-admin/admin.php?page=jetpack-debug-tools
* Enable the Autoloader module.
* Visit the /wp-admin/admin.php?page=autoloader-debug-helper page.
* Make sure you see all Jetpack plugins listed.
* Make sure you see the files and classes available from the classmaps.
* Play around with the `define( 'JETPACK_AUTOLOAD_DEV', true );` constant, see that the versions are correctly reflected in the classmap data. Bonus points for trying with dev/prod versions of plugins.
* Try to rename a file from one of the classes and see that the debugger correctly reports it.
